internal/server/evaluator.go

```diff
@@ -68,9 +78,12 @@ func (s *Server) batchEvaluate(ctx context.Context, r *flipt.BatchEvaluationRequ
 
 	// TODO: we should change this to a native batch query instead of looping through
 	// each request individually
-	for _, flag := range r.GetRequests() {
+	for _, req := range r.GetRequests() {
+		// ensure all requests have the same namespace
+		req.NamespaceKey = r.NamespaceKey
```

Comment:
My gut says we probably should just reject a batch request if it contains get requests with namespaces which don't match the batch payload namespace.

Maybe we could at-least support implying a blank namespace in a child evaluate means inherits the parent batch request namespace?

What do you think?

Reply:
yeah that makes sense. in retrospect I should have made the batch API use it's own set of request objects which don't share the same properties with the `EvaluateRequest`.. but live and learn I guess. something to think about if we do a v2 API

Reply:
Yeah that makes sense.